pkg/daemon/ceph/osd/daemon.go

```diff
@@ -66,12 +66,12 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 		go handleTerminate(context, lvPath, volumeGroupName)
 
 		if err := context.Executor.ExecuteCommand(false, "", "vgchange", "-an", volumeGroupName); err != nil {
-			return fmt.Errorf("failed to deactivate volume group for lv %+v. Error: %+v", lvPath, err)
+			return fmt.Errorf("failed to deactivate volume group for lv %q. %+v", lvPath, err)
 		}
+	}
 
-		if err := context.Executor.ExecuteCommand(false, "", "vgchange", "-ay", volumeGroupName); err != nil {
-			return fmt.Errorf("failed to activate volume group for lv %+v. Error: %+v", lvPath, err)
-		}
+	if err := context.Executor.ExecuteCommand(false, "", "vgchange", "-ay", volumeGroupName); err != nil {
```

Comment:
The volumeGroupName won't be set if it's not a PVC, if you could take another look per discussion